models/RelatedTransaction.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 Coinbase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+description: |
+  The related_transaction allows implementations to link together multiple transactions.
+type: object
+required:
+  - transaction_identifier
+  - direction
+properties:
+  transaction_identifier:
+    $ref: 'TransactionIdentifier.yaml'
+  network_identifier:
+    $ref: 'NetworkIdentifier.yaml'
+  direction:
+    description: |
+      Direction of the relation (i.e. cross-shard/cross-network sends may reference "back" to an earlier transaction
+      and async execution may reference "forward")
+    type: string
+    enum:
```

Comment:
I'd define this ENUM as its own model (this makes the go codegen much nicer): https://github.com/coinbase/rosetta-specifications/pull/66/files#diff-9c2b381a678e90c4774820b58cffb4416b18325f069fcc94e62a6c1333996ee2R1